docs/tutorial.md

```diff
@@ -0,0 +1,396 @@
+deploy to it.
+
+I will assume familiarity with git itself.
+
+## First things first
+
+You will need certain tools to be installed in order to use the tools
+in the repository template.  They are:
+
+- The gnu coreutils.
+- Gnu make.
+
+These are some of the tools that the engineering community assumes you
+will have for compatibility with a wide range of projects and
+products, so install them if you haven't already.
+
+Some other tools will be installed by the framework if you don't
+already have them installed.  I'll call those out as and when they
+come up.
+
+I do not assume, dictate, or otherwite hint at the correct text
+editor for you to be using.  I just silently judge you for your
+choice.
+
+## Cloning the template
+
+The first step to your brand new app is to clone the repository.  For
+the purposes of this tutorial, we're going to clone it into your
+personal account rather than into any of the NHS github
+organisations. That's not what you'd be doing for anything we want to
+go into production, but it means you can keep it as a reference and we
+don't need to worry about naming collisions.
+
+Go to <https://github.com/nhs-england-tools/repository-template> in your
+browser, and click the green button marked `Use this template`. From
+the dropdown that appears, select `Create a new repository`.
+
+You will be taken to github's `Create a new repository` page.  You
+might notice that this is the same page as github shows you when you
+create an ordinary repository in the web user interface, but pay
+attention to the `Repository template` field: you will see that it is
+pre-populated with the `nhs-england-tools/repository-template` value.
+That is as we want, so leave it as is.
+
+Leave the `Include all branches` checkbox unticked.
+
+Under `Owner`, select yourself from the `Choose an owner` dropdown.
+Visibility options will appear under the Description field.  The
+repository is MIT-licensed, and we're working in the open, so you
+don't need to worry about this being a public repository.  You can
+make it private if you prefer.
+
+Under `Repository name`, give the repository a name of
+`nhset-hello-world`.
+
+[TODO: This means the SonarCloud integration won't work, but we can't
+get around that without being able to make repositories in a different
+org.]
+
+If a description will help you remember the purpose of this project in
+the future, put something meaningful to you in the `Description`
+field.
+
+Click the green `Create repository` button, and github will show you a
+loading page while your new instance of the repository is created.
+
+Now that you have your repository, clone it to your local machine for
+the next step.
+
+## Python time
+
+For our Python application, we will be building an API endpoint which
+returns a JSON blob that looks like this:
+
+```json
+{
+  "message": "Hello World"
+}
+```
+
+The framework contains a mechanism for ensuring that your local
```

Comment:
As per our recent decision not to install `asdf` by default, we may need to reword this section.

Reply:
I've given it a go; it feels like we'll be revisiting some of this fairly regularly anyway, but I think it's workable now.